Formula/php.rb

```diff
@@ -0,0 +1,368 @@
+      if ext_config_path.exist?
+        inreplace ext_config_path,
+          /#{extension_type}=.*$/, "#{extension_type}=#{php_ext_dir}/#{e}.so"
+      else
+        # Do not load snmp extension by default
+        extension_type.prepend(";") if e == "snmp"
+        ext_config_path.write <<~EOS
+          [#{e}]
+          #{extension_type}="#{php_ext_dir}/#{e}.so"
+        EOS
+      end
+    end
+  end
+
+  def php_version
+    version.to_s.split(".")[0..1].join(".")
+  end
+
+  plist_options :startup => true, :manual => "php-fpm"
+
+  def plist; <<~EOPLIST
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <true/>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_sbin}/php-fpm</string>
+          <string>--nodaemonize</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/php-fpm.log</string>
+      </dict>
+    </plist>
+    EOPLIST
+  end
+
+  test do
+    system "#{bin}/php", "-i"
+    system "#{sbin}/php-fpm", "-t"
+    system "#{bin}/phpdbg", "-V"
+    system "#{bin}/php-cgi", "-m"
+    shell_output("#{bin}/php -r '(extension_loaded(\"Zend OPCache\"))?exit(0):exit(1);'")
+    begin
+      require "socket"
+      
+      server = TCPServer.new(0)
+      port = server.addr[1]
+      server.close
+
+      expected_output = /^Hello world!$/
+      (testpath/"index.php").write <<~EOS
+        <?php
+        echo 'Hello world!';
+      EOS
+      main_config = <<~EOS
+        Listen #{port}
+        ServerName localhost:#{port}
+        DocumentRoot "#{testpath}"
+        ErrorLog "#{testpath}/httpd-error.log"
+        ServerRoot "#{Formula["httpd"].opt_prefix}"
+        LoadModule authz_core_module lib/httpd/modules/mod_authz_core.so
+        LoadModule unixd_module lib/httpd/modules/mod_unixd.so
+        LoadModule dir_module lib/httpd/modules/mod_dir.so
+        DirectoryIndex index.php
+      EOS
+
+      (testpath/"httpd.conf").write <<~EOS
+        #{main_config}
+        LoadModule mpm_prefork_module lib/httpd/modules/mod_mpm_prefork.so
+        LoadModule php7_module #{lib}/httpd/modules/libphp7.so
+        <FilesMatch \.(php|phar)$>
+          SetHandler application/x-httpd-php
+        </FilesMatch>
+      EOS
+
+      (testpath/"httpd-fpm.conf").write <<~EOS
+        #{main_config}
+        LoadModule mpm_event_module lib/httpd/modules/mod_mpm_event.so
+        LoadModule proxy_module lib/httpd/modules/mod_proxy.so
+        LoadModule proxy_fcgi_module lib/httpd/modules/mod_proxy_fcgi.so
+        <FilesMatch \.(php|phar)$>
+          SetHandler "proxy:fcgi://127.0.0.1:9000"
```

Comment:
Is there a reason to hard code 9000?

Reply:
Only that it is the default in fpm. `By default, php-fpm will respond to CGI requests listening on localhost http port 9000.` - we should be able to change this to a dynamic port by creating a specific ad-hoc config.